awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/OtlpUdpSpanExporter.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.opentelemetry.javaagent.providers;
+
+import io.opentelemetry.exporter.internal.otlp.traces.TraceRequestMarshaler;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Exports spans via UDP, using OpenTelemetry's protobuf model. The protobuf modelled spans are
+ * Base64 encoded and prefixed with AWS X-Ray specific information before being sent over to {@link
+ * UdpSender}.
+ *
+ * <p>This exporter is NOT meant for generic use since the payload is prefixed with AWS X-Ray
+ * specific information.
+ */
+@Immutable
+class OtlpUdpSpanExporter implements SpanExporter {
+
+  private static final Logger logger = Logger.getLogger(OtlpUdpSpanExporter.class.getName());
+
+  // The protocol header and delimiter is required for sending data to X-Ray Daemon or when running
+  // in Lambda.
+  // https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-daemon
+  private static final String PROTOCOL_HEADER = "{\"format\": \"json\", \"version\": 1}";
+  private static final char PROTOCOL_DELIMITER = '\n';
+
+  // These prefixes help the backend identify if the spans payload is sampled or not.
+  private static final String FORMAT_OTEL_SAMPLED_TRACES_BINARY_PREFIX = "T1S";
+  private static final String FORMAT_OTEL_UNSAMPLED_TRACES_BINARY_PREFIX = "T1U";
+
+  private final AtomicBoolean isShutdown = new AtomicBoolean();
+
+  private final UdpSender sender;
+  private final boolean sampled;
+
+  OtlpUdpSpanExporter(UdpSender sender, boolean sampled) {
+    this.sender = sender;
+    this.sampled = sampled;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<SpanData> spans) {
+    if (isShutdown.get()) {
+      return CompletableResultCode.ofFailure();
+    }
+
+    TraceRequestMarshaler exportRequest = TraceRequestMarshaler.create(spans);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    try {
+      exportRequest.writeBinaryTo(baos);
+      String payload =
+          PROTOCOL_HEADER
+              + PROTOCOL_DELIMITER
+              + (sampled
```

Comment:
Combining the discussion at [this thread](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/944/files#r1829838239), my suggestion for you to evaluate would be

Option 1. Read the sampled decision from span if you want the exporter to work independently.
Option 2. Pass the predefined (un)sampled protocol prefix into exporter instead of a boolean `sampled`, and add comment on top with some description. I don't see why we should repeatedly build the same string in the exporting function.

Reply:
I had a discussion with @wangzlei and he explained that the idea of using 2 processors is to segregate sampled and unsampled spans into respective batches for export. This ensures each payload contains either all sampled or unsampled spans. 

Given how the exporter receives a collection of spans in its `export()`, I think something like Option 2 (assuming the 2 BSP solution works in Java) would be better than having to separate spans in the exporter and send two payloads.

Reply:
@bjrara - BSP has [this flag](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorBuilder.java#L45) which enables exporting of unsampled spans too.
We can have a custom SpanProcessor delegate the unsampled spans to a BSP with `exportUnsampledSpans` enabled.

Reply:
Did a quick and dirty test to validate the above works.

#### Setup
##### Custom SpanProcessor `AwsUnsampledSpanProcessor`:
```java
package software.amazon.opentelemetry.javaagent.providers;

import io.opentelemetry.context.Context;
import io.opentelemetry.sdk.common.CompletableResultCode;
import io.opentelemetry.sdk.trace.ReadWriteSpan;
import io.opentelemetry.sdk.trace.ReadableSpan;
import io.opentelemetry.sdk.trace.SpanProcessor;
import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
import io.opentelemetry.sdk.trace.export.SpanExporter;

public class AwsUnsampledSpanProcessor implements SpanProcessor {
    public final SpanProcessor forwardingBSP;

    AwsUnsampledSpanProcessor(SpanExporter exporter) {
        this.forwardingBSP = BatchSpanProcessor.builder(exporter)
                .setExportUnsampledSpans(true)
                .build();
    }

    @Override
    public void onStart(Context context, ReadWriteSpan readWriteSpan) {
        forwardingBSP.onStart(context, readWriteSpan);
    }

    @Override
    public boolean isStartRequired() {
        return forwardingBSP.isStartRequired();
    }

    @Override
    public void onEnd(ReadableSpan readableSpan) {
        if (readableSpan.getSpanContext().isSampled()) {
            return;
        }
        forwardingBSP.onEnd(readableSpan);
    }

    @Override
    public boolean isEndRequired() {
        return forwardingBSP.isEndRequired();
    }

    @Override
    public CompletableResultCode shutdown() {
        return forwardingBSP.shutdown();
    }

    @Override
    public CompletableResultCode forceFlush() {
        return forwardingBSP.forceFlush();
    }

    @Override
    public void close() {
        forwardingBSP.close();
    }
}
```
##### Code to create a unsampled span and process via the `AwsUnsampledSpanProcessor`
```java
@Test
  public void testExportingUnsampledSpansWithCustomProcessor() {

    SpanExporter unsampledSpanExporter = new OtlpUdpSpanExporterBuilder()
            .setPayloadSampleDecision(TracePayloadSampleDecision.UNSAMPLED)
            .build();
    SpanProcessor unsampledSpanProcessor = new AwsUnsampledSpanProcessor(unsampledSpanExporter);

    SdkTracerProvider tp = SdkTracerProvider.builder()
            .addSpanProcessor(unsampledSpanProcessor)
            .setSampler(AlwaysRecordSampler.create(Sampler.alwaysOff())) //set sampler
            .build();
    Tracer tracer = OpenTelemetrySdk.builder().setTracerProvider(tp).build().getTracer("hello");

    Span mySpan = tracer.spanBuilder("DoTheLoop_8").startSpan();
    int numIterations = 5;
    mySpan.setAttribute("NumIterations", numIterations);
    mySpan.setAttribute(AttributeKey.stringArrayKey("foo"), Arrays.asList("bar1", "bar2"));
    try (var scope = mySpan.makeCurrent()) {
      for (int i = 1; i <= numIterations; i++) {
        System.out.println("i = " + i);
      }
    } finally {
      mySpan.end();
    }

    unsampledSpanProcessor.forceFlush();
  }
```

#### Observations
- When spans are unsampled the payload is exported with `T1U` prefix
- When changing the sampler to `alwaysOn`, no spans are exported. This is expected since the only SpanProcessor added in my setup is the `AwsUnsampledSpanProcessor` so the sampled spans are simply dropped.
- This shows that the `AwsUnsampledSpanProcessor` can be used to separate and export only the unsampled spans.